lib/problems.rb

```diff
@@ -1,13 +1,40 @@
-require_relative './stack.rb'
+require_relative './Stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(nlogn)
+# Space Complexity: O(n)
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  stack = Stack.new
+  arr = string.split('')
+  chars = ['(', '[', '{']
+  arr.each do |char|
+    if chars.include?(char)
+      stack.push(char)
+    else
```

Comment:
Could you do this with a hash where the keys are the open brace and he values are the matching close brace?  That might help you dry up those if-else statements.